ast/desugar/Desugar.cc

```diff
@@ -706,6 +706,44 @@ class DuplicateHashKeyCheck {
     }
 };
 
+// Returns the flattened key/value pairs from the Kwargs Hash, if there was one.
+// If Kwargs Hash contains any splats, we skip the flattening.
+//
+// Returns nullopt if there was no Kwargs Hash
+optional<parser::NodeVec> flattenKwargs(unique_ptr<parser::Hash> kwargsHash) {
```

Comment:
This was a big pretty chunk of the duplication. Now it's only called once, but pull it out here really helps readability.

We drop from 12 levels of indentation (!!!) to "only" 8.